File: exercise_py/exercise_0.py
#!/usr/bin/env python

from sympy import sieve
import time


def primality_test_simple(n):
    """素数判定（単純な実装）"""
    
    n = int(n)
    if n < 2:
        return False
    else:
        for i in range(2, n, 1):
            if n % i == 0:
                return False
    return True

def do_2(start, end, test_func):
    """素直な実装"""
    
    t1 = time.time()
    start = int(start)
    end = int(end)
    prime_list = []
    print("start ...")
    
    for n in range(start, end+1, 1):
        if test_func(n):
            prime_list.append(n)

    print('prime number is ...')
    #print(prime_list)
    print("Total = ", len(prime_list))
    print("time = ", time.time() - t1)
    return prime_list


def primality_test_using_sieve(n):
    """素数判定（sieve使用）
    
    本来の使い方じゃないので遅いかも
    """
    
    n = int(n)
    if n in sieve:
        return True
    else:
        return False



if __name__ == '__main__':

    #print(primality_test_simple(999961))  # 100万以下で最大の素数
    do_2(2, 34678, primality_test_using_sieve)

File: exercise_jl/exercise_0.jl

"""
素数判定
"""
function is_prime_simple(n::Int64)
    if n < 2
        return false
    elseif n == 2
        return true
    else
        for i in 2:n-1
            #println(i, "...")
            if n % i == 0
                return false
            end
            #println("わりきれない")
        end
    end
    return true
end

function do_2(n₀::Int64, n₁::Int64)
    ns = n₀:n₁
    prime_list = is_prime_simple.(ns)
    num = prime_list' * prime_list
    println(n₀, " から ", n₁ ," に素数は", num, "個ある")
end



#is_prime_simple(999961)
#@time is_prime_simple(999961)  # 100万以下で最大の素数


#do_2(1, 10)
@time do_2(1, 1000)